Mage.Sets/src/mage/cards/s/SanctumOfShatteredHeights.java

```diff
@@ -0,0 +1,56 @@
+package mage.cards.s;
+
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.DiscardTargetCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.dynamicvalue.common.PermanentsOnBattlefieldCount;
+import mage.abilities.effects.common.DamageTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.Zone;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.predicate.Predicates;
+import mage.target.common.TargetCardInHand;
+
+import java.util.UUID;
+
+/**
+ * @author jmharmon
+ */
+
+public final class SanctumOfShatteredHeights extends CardImpl {
+
+    private static final FilterControlledPermanent filter = new FilterControlledPermanent("Shrine");
+    private static final FilterCard filter1 = new FilterCard("a land card or Shrine card");
+
+    static {
+        filter.add(SubType.SHRINE.getPredicate());
+        filter1.add(Predicates.or(CardType.LAND.getPredicate(), SubType.SHRINE.getPredicate()));
+    }
+
+    public SanctumOfShatteredHeights(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{R}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.SHRINE);
+
+        // {1}, Discard a land card or Shrine card: Sanctum of Shattered Heights deals X damage to target creature or planeswalker, where X is the number of Shrines you control.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new DamageTargetEffect(new PermanentsOnBattlefieldCount(filter)), new ManaCostsImpl("{1}"));
```

Comment:
you don't need the zone in the constructor but it's not that important, also you need to set the text since it won't autogenerate correctly

Reply:
Add ability text. I think it's in the correct spot. There's something wrong with my Intellij that won't let me wrong the server to view the card in the client.